GameObject.hpp

```diff
@@ -44,7 +47,16 @@ namespace spic {
              */
             template<class T>
             static std::shared_ptr<GameObject> FindObjectOfType(bool includeInactive = false) {
-                // ... implementation here
+                for(GameObject gameObject : administration){
+                    if(typeid(GameObject) == typeid(T)){
+                        if(includeInactive){
+                            return std::make_shared<GameObject>(gameObject);
+                        } else if(gameObject.Active()){
+                            return std::make_shared<GameObject>(gameObject);
+                        }
+                    }
+                }
```

Comment:
Two comments about this code snippet:
- I'm not a fan of those nested if statements. You can fix this by using early returns/continues.
- In your nested if else statement, you are checking two different things but you do the same thing either way. Therefore, you can simply merge the if statements into one.

Here's how I would've written your code:
```cpp
for(GameObject gameObject : administration) {
	if (typeid(GameObject) != typeid(T))
		continue;
		
	if (includeInactive || gameObject.Active()) {
		return std::make_shared<GameObject>(gameObject);
	}
}
```